identity/mock-iot-server/src/dps.rs

```diff
@@ -9,13 +9,12 @@ fn register(
     context: &mut crate::server::Context,
 ) -> Response {
     let body = if let Some(body) = body {
-        let body: aziot_cloud_client_async::DpsRequest::TpmRegistration =
-            match serde_json::from_str(body) {
-                Ok(body) => body,
-                Err(_) => return Response::bad_request("failed to parse register body"),
-            };
+        let body: serde_json::value::Value = match serde_json::from_str(body) {
```

Comment:
It was made to allow for the code added on line 45:
`request_payload: body["payload"].clone(),`

Before the change, the `body` was of type `TpmRegistration`, which doesn't have a `payload` field. With the change, `body` is the more generic `serde_json::Value` type, which can be used ot access `TpmRegistration` or `DeviceRegistration` fields.

I could of course revert the change on this line and add another block that retrieves the `DeviceRegistration` object, but I preferred this approach due to the minimal changes it required.

Reply:
>Before the change, the body was of type `TpmRegistration`, which doesn't have a `payload` field.

Yes, because you didn't add it. The point of your change is to add support for payloads, yes? So wouldn't you add it to the `TpmRegistration` struct as well?

Reply:
Yes, good catch. I misinterpreted device registration as separate from tpm registration, but looking at the DPS API doc again, I see that tpm registration is a part of device registration: https://docs.microsoft.com/en-us/rest/api/iot-dps/device/runtime-registration/register-device#deviceregistration

I'll make the necessary changes.

Reply:
Actually, looking at the following code, it looks like for TPM registration, there are two calls to the device registration endpoint. The first is via the call to `get_tpm_nonce()`, which sends a `TpmRegistration` object. The second sends the `DeviceRegistration` object, which will contain the payload. With that said, it's not clear to me whether first call should include the payload. Thoughts?

https://github.com/Azure/iot-identity-service/blob/4de8beaab82f28bfef7a52ddbaf9a6f3c532badb/identity/aziot-cloud-client-async/src/dps/mod.rs#L165-L195

Reply:
In your PR description you posted https://docs.microsoft.com/en-us/azure/iot-dps/how-to-send-additional-data . The first example there is of a payload in a TpmRegistration. But in any case that is something you should ask DPS to confirm.

Reply:
I checked with DPS and they said the payload is not currently used in the initial call, but since that is undocumented behavior, it's recommended to send it in both calls. I've updated the code accordingly.